tools/hidl/hidlcomp.cpp

```diff
@@ -5373,57 +5374,166 @@ void EspServInfo::write_factory_impl()
     outf(" IClient%s * create%sClient() {  return new CClient%s(); }\n", name_, name_, name_);
 }
 
+const char * translateAuthLevel(const char * level)
+{
+    /*
+     *  This method might belong in seclib where the enumeration is defined (selib.h)
+     *  enum SecAccessFlags
+     *  {
+     *     SecAccess_Unknown = -255,
+     *     SecAccess_None = 0,
+     *     SecAccess_Access = 1,
+     *     SecAccess_Read = 3,
+     *     SecAccess_Write = 7,
+     *     SecAccess_Full = 255
+     *   };
+    */
+
+    if (!level || !*level)
+    {
+        outs(2, "\n//FEATURE LEVEL NOT SET, DEFAULTING TO 'READ'\n");
+        return "SecAccess_Read";
+    }
+
+    if (strieq(level, "NONE"))
+    {
+        outs(2, "\n//WARNING: FEATURE LEVEL AUTHORIZATION HAS BEEN TURNED OFF!!\n");
+        return "SecAccess_None";
+    }
+
+    if (strieq(level, "INMETHOD"))
```

Comment:
Not sure I like the term "INMETHOD".  Could be convinced, but what do you think of "DELAYED" or "INLINE"?